module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -306,4 +307,44 @@ public function updateResource($entity, $newResource, $oldResource)
         $query->setParameters($parameters);
         $query->execute();
     }
+
+    /**
+     * Apply a sort parameter to a query on the resource table.
+     *
+     * @param string $query Query to modify
+     * @param string $sort  Field to use for sorting (may include
+     *                      'desc' qualifier)
+     * @param string $alias Alias to the resource table (defaults to
+     *                      'resource')
+     *
+     * @return string
+     */
+    public static function applySort($query, $sort, $alias = 'r')
```

Comment:
I don't think there's really a need to pass in the $query parameter here. Would it be simpler to call this getOrderByClause and return just the ORDER BY clause (or an empty string if none is needed)? Then you could just concatenate this on to a query in the calling code.